apps/dot_ext/models.py

```diff
@@ -221,16 +221,38 @@ class ExpiresIn(models.Model):
 
 
 def archive_token(sender, instance=None, **kwargs):
-    archived_token = ArchivedToken()
     tkn = instance
-    archived_token.user = tkn.user
-    archived_token.token = tkn.token
-    archived_token.application = tkn.application
-    archived_token.expires = tkn.expires
-    archived_token.scope = tkn.scope
-    archived_token.created = tkn.created
-    archived_token.updated = tkn.updated
-    archived_token.save()
+    ArchivedToken.objects.get_or_create(
+        user=tkn.user,
+        token=tkn.token,
+        application=tkn.application,
+        expires=tkn.expires,
+        scope=tkn.scope,
+        created=tkn.created,
+        updated=tkn.updated,
+    )
+
+
+def application_post_delete(sender, instance=None, **kwargs):
```

Comment:
I understand why this is necessary, but it's necessity means something isn't behaving in an expected way. I'm looking into alternatives.

Reply:
I think that you're right. Was also looking in to alternatives without any luck, and with the idea of not changing or adding to the application/user models.  I'll also continue looking in to alternatives. Let me know what you find out. TY!

Reply:
TY for the alternate solution! This has tested out well and no more errors.